diff --git a/node_modules/@adminjs/sql/lib/dialects/postgres.parser.js b/node_modules/@adminjs/sql/lib/dialects/postgres.parser.js
index ea43916..a009f43 100644
--- a/node_modules/@adminjs/sql/lib/dialects/postgres.parser.js
+++ b/node_modules/@adminjs/sql/lib/dialects/postgres.parser.js
@@ -132,7 +132,11 @@ export class PostgresParser extends BaseDatabaseParser {
         where attrelid = c.conrelid and ARRAY[attnum] <@ c.conkey) as col, 
         (select r.relname from pg_class r where r.oid = c.confrelid) as referenced_table
       from pg_constraint c
-      where c.conrelid = (select oid from pg_class where relname = '${table}')
+      where c.conrelid = (
+        select r.oid from pg_class r
+        join pg_namespace n on n.oid = r.relnamespace
+        where r.relname = '${table}' and n.nspname = '${schemaName}'
+      )
       and (select r.relname from pg_class r where r.oid = c.confrelid) is not null
     `);
         const relations = await relQuery;
